src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -1097,9 +1096,8 @@ pub extern "C" fn materialize_directories(
         &[externs::store_tuple(&entries)],
       );
       output
-    });
-
-    scheduler.core.executor.spawn_on_io_pool(work_future).wait()
+    })
+    .wait()
```

Comment:
Why is this changing from blocking to not? Is it because the blocking-ness is pushed down onto the actual-blocking parts? IIRC last time I tried to do that, there was a massive perf regression, which is why this is how it was... 🤞

Reply:
This callsite was redundant: `wait()` on a 0.1 future is itself a blocking method that uses the current thread as the executor for the future. It's gone in `stdlib` futures in favor of explicit calls to `${MyRuntime}.block_on(..)`.

Reply:
Oooor I haven't run tests that have covered this yet, and it will complain about one of the nested `spawn` calls not occurring in the context of a `Runtime`! We shall see, heh. The patch currently runs `./pants list` successfully, but some of the rust tests haven't been ported, and there are likely to be a few "no Runtime" or "too many Runtimes" type errors at various callsites.